docs/unified/hardkernel/initial-deployment.md

```diff
@@ -0,0 +1,36 @@
+# Initial deployment
+
+Initial flashing of Dasharo firmware can be done from Linux using flashrom with
+the internal programmer. This document describes the process of building,
+installing and running flashrom on Ubuntu 24.04.
+
+## Build flashrom
+
+Please follow generic guide for [Dasharo flashrom fork](../../osf-trivia-list/deployment.md#how-to-install-dasharo-flashrom-fork).
+
+## Reading flash contents
+
+Always prepare a backup of the current firmware image. To read from the flash
+and save it to a file (`dump.rom`), execute the following command:
+
+```bash
+flashrom -p internal -r dump.rom
+```
+
+If you forgot to do this, you can get the original, proprietary
+firmware from the [ODROID wiki website](https://wiki.odroid.com/odroid-h4/hardware/h4_bios_update#bios_release)
+ to restore it in case it is bricked. Using the firmware other than
+your own backup should be the last resort.
+
+## Flashing Dasharo
+
+=== "ODROID H4+"
+
+    To flash Dasharo on the platform, execute the following command - replace `[path]`
+    with the path to the Dasharo image you want to flash, e.g. `hardkernel_odroid_h4_v0.9.0.rom`.
+
+    ```bash
+    sudo flashrom -p internal -w [path] --ifd -i bios
```

Comment:
Only flashing bios works for now, when our fimware image does not include Intel blobs. This might change.

Reply:
Initial deployment will never work with flashing full binary. ME cannot be disabled on proprietary firmware, so it will always require flashing BIOS region. Unless someone will power cycle the board (but still unsure if it will work).

Reply:
@miczyg1 are you sure it can't be disabled? From what I can see ME is being disabled when following recovery instructions from ODROID: https://wiki.odroid.com/odroid-h4/hardware/restore_h4_bios

Windows:

> Step 3. Make disable ME
Go to location where FPT(Flash Programming Tool) file in there and run FPTW64.exe -disableme

EFI shell:

> Step 3. Make disable ME
Go to location where FPT(Flash Programming Tool) file in there and run fpt -disableme

Reply:
Yes, it can't (no setup option for it). Unless you run those tools from the wiki. But relying on those tools is problematic as well:

1. You don't always have the tools provided by BIOS vendor.
2. The method with FPTW64 depends on the firmware to not lock the ME disabling mechanism.
3. Each ME disabling mechanism requires a reset of the platform, so I am not sure how ME disabling here is supposed to even work.
4. Each board may have different toolset for this, which would make the documentation/procedures less generic. 
5. If something doesn't work, you have no means to fix it.